### PR TITLE
NEML time step limit and fix for infinite strain rate FPE

### LINKED ISSUE #167: Add material time step limit to NEML
## Reason
Cap time step based on a maximum inelastic strain increment.

## Design
Add `matl_timestep_limit` (should be named `material_timestep_limit`) to the NEML material. For now just compute it from the norm of the difference between current and previous inelastic strain.  

## Impact
Added capability

### PR DESCRIPTION
**Pull request by dschwen**
Closes #167

### CODE REVIEW

File: contrib/neml.mk [hunk 1]
```diff
@@ -25,7 +25,7 @@ $(NEML_DIR)/src/%.$(obj-suffix) : $(NEML_DIR)/src/%.cxx
 
 ADDITIONAL_INCLUDES  += -I$(neml_includes) $(neml_dep_includes)
 ADDITIONAL_LIBS      += -L$(NEML_DIR) -lneml-$(METHOD)
-ADDITIONAL_CPPFLAGS  += -DNEML_ENABLED
+ADDITIONAL_CPPFLAGS  += -DNEML_ENABLED -DNEML_STRAIN_RATE_LIMIT=1e10
```
Comment (lynnmunday) [inline contrib/neml.mk:28]: what does this new flag do?

File: include/materials/NEMLStressBase.h [hunk 2]
```diff
@@ -62,6 +62,11 @@ class NEMLStressBase : public ComputeStressBase
   /// Inelastic strain tensor
   MaterialProperty<RankTwoTensor> & _inelastic_strain;
 
+  const bool _compute_dt;
+  const Real _target_increment;
+  const MaterialProperty<RankTwoTensor> * _inelastic_strain_old;
+  MaterialProperty<Real> * _material_dt;
```
Comment (lynnmunday) [inline include/materials/NEMLStressBase.h:68]: are these pointers so that you can set them to nullptr when no compute_dt is defined?

**Review by lynnmunday — APPROVED** [review]
Just comments for my own self improvement.   Is this the only material to compute its own critical timestep size?  I think other materials only compute an inelastic strain increment and then use that in the materialTimestepper postprocessor to cut the timestep.  I guess you can choose either way now that neml models will output the inelastic_strain.  Is there a reason critical timestep makes more sense?  Will NEML creep models also return their inelastic strain increment or do that compute an inelastic strain increment?

File: contrib/neml.mk (hunk 1 above)
Comment (dschwen) [reply contrib/neml.mk:28]: It is needed as a threshold value for inelastic strain rata are considered "too large" (i.e will result in floating point errors)

File: include/materials/NEMLStressBase.h (hunk 2 above)
Comment (dschwen) [reply include/materials/NEMLStressBase.h:68]: yes

**Comment by dschwen** [comment]
> Just comments for my own self improvement. Is this the only material to compute its own critical timestep size? 
No, there are a bunch more (just search for `material_timestep_limit` in MOOSE.

> I think other materials only compute an inelastic strain increment and then use that in the materialTimestepper postprocessor to cut the timestep. I guess you can choose either way now that neml models will output the inelastic_strain. Is there a reason critical timestep makes more sense?
Not to my knowledge. AFAIK material timestepping always uses `material_timestep_limit`

> Will NEML creep models also return their inelastic strain increment or do that compute an inelastic strain increment?
This is currently not exported form NEML

**Pull request merged**